pkg/apis/v1alpha1/dnspolicy_types.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2023 The MultiCluster Traffic Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DNSPolicySpec defines the desired state of DNSPolicy
+type DNSPolicySpec struct {
+
+	// +kubebuilder:validation:Required
+	// +required
+	DNSRecordRef DNSRecordRef `json:"dnsRecordRef"`
+
+	// +optional
+	HealthCheck HealthCheckSpec `json:"healthCheck,omitempty"`
+}
+
+// DNSPolicyStatus defines the observed state of DNSPolicy
+type DNSPolicyStatus struct {
+
+	// conditions are any conditions associated with the policy
+	//
+	// If configuring the policy fails, the "Failed" condition will be set with a
+	// reason and message describing the cause of the failure.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// observedGeneration is the most recently observed generation of the
+	// DNSPolicy.  When the DNSPolicy is updated, the controller updates the
+	// corresponding configuration. If an update fails, that failure is
+	// recorded in the status condition
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].status",description="DNSPolicy ready."
+
+// DNSPolicy is the Schema for the dnspolicies API
+type DNSPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DNSPolicySpec   `json:"spec,omitempty"`
+	Status DNSPolicyStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// DNSPolicyList contains a list of DNSPolicy
+type DNSPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DNSPolicy `json:"items"`
+}
+
+type ProviderHealthCheckSpec struct {
```

Comment:
Do we see that as something we would expose externally? I am thinking we would likely try to present a single API and take care of the messy translations internally?

Reply:
I'm thinking of the possibility of a different API, as in a separate set of health checks that don't translate to health checks in Route 53. I don't know exactly how that would look like but I thought having a separate section would be a good first step. If you think that's not going to happen I can simplify this

Reply:
I think for now and as this is a alpha api, I would be inclined to simply make the spec what is in ProviderHealthCheckSpec. We can make it more complicated in the future if the need shows itself. Lets just ensure we mark this an alpha api that way the expectation is it might change significantly

Reply:
Cool, I updated the PR with these changes